dataset_builders/pie/scientific_papers/scientific_papers.py

```diff
@@ -0,0 +1,81 @@
+import dataclasses
+from typing import Any, Dict, List
+
+import datasets
+from pytorch_ie.core import annotation_field
+from pytorch_ie.documents import TextBasedDocument
+
+from pie_datasets import GeneratorBasedBuilder
+
+
+@dataclasses.dataclass
+class ScientificPapersDocument(TextBasedDocument):
+    """A PIE document for scientific papers dataset."""
+
+    abstract: str = annotation_field()
+    section_names: List[str] = annotation_field()
+
+
+def example_to_document(
+    example: Dict[str, Any],
+) -> ScientificPapersDocument:
+    """Convert a Huggingface Scientific Papers example to a PIE document."""
+    document = ScientificPapersDocument(
+        text=example["article"],
+    )
+    document.abstract = example["abstract"]
+    document.section_names = example["section_names"].split("\n")
```

Comment:
`AnnotationLayer`s should get their content with either `.append(new_annotation)` or `extend(list_of_new_annotations)`